lib/LaTeXML/Package/physics.sty.ltxml

```diff
@@ -341,37 +355,95 @@ DefMacro('\lx@physics@diff{}{}{}', sub {
     my ($gullet, $cs, $semantic, $function) = @_;
     my $cfunc = Invocation(T_CS('\@CSYMBOL'), $semantic);
     my $pfunc = $function;
-    if (my $power = $gullet->readOptional) {
-      ($cfunc, $pfunc) = phys_applyPower($cfunc, $pfunc, $power); }
-    my ($arg, $open, $close) = phys_readArg($gullet, 0, undef, undef, '(' => T_OTHER(')'));
-    return ($arg
-      ? phys_applyFunction($cfunc, $pfunc, 1, $open, $close, $arg)
-      : Invocation(T_CS('\DUAL'), undef, $cfunc, Invocation(T_CS('\@WRAP'), $pfunc))); });
-
+    my $power = $gullet->readOptional;
+    my ($arg, $open, $close) = phys_readArg($gullet, 0, '(' => T_OTHER(')'));
+    # Note: functional-power!
+    if ($power) {
+      $cfunc = Invocation(T_CS('\@APPLY'),
+        Tokens(Invocation(T_CS('\@CSYMBOL'), T_OTHER('power')), Tokens($cfunc, T_XMArg(2))));
+      $pfunc = Tokens($pfunc, Invocation(T_CS('\@@POSTSUPERSCRIPT'), T_XMArg(2))) }
+    return BuildXMDual(
+      { ($arg ? () : (role => 'DIFFOP')),
+        reversion => Tokens($cs,
+          ($power ? (T_OTHER('['), T_XMArg(2), T_OTHER(']')) : ()),
+          ($arg ? phys_revArg(T_XMArg(1), $open, $close) : ())) },
+      ($arg ? Invocation(T_CS('\@APPLY'), Tokens($cfunc, $arg)) : $cfunc),
+      Tokens($pfunc, phys_open(1, $open), ($arg ? T_XMArg(1) : ()), phys_close(1, $close)),
+      $arg, ($power ? $power : ())); });
+
+# Trys to be (too) many things (but the syntax is insufficient)
+# \derivative{var}
+# \derivative[n]{var}
+# \derivative{expr}{var}
+# \derivative[n]{expr}{var}
+# \pderivative{expr}{var1}{var2}
+# \derivative{var}(expr)
+# \derivative[n]{var}(expr)
+# BUT
+# \pderivative[n]{expr}{var1}{var2}  n is read but IGNORED
+# \pderivative{expr}{var1}{var2}(expr2)  expr2 is read but IGNORED
 DefMacro('\lx@physics@deriv{}{}{}', sub {
     my ($gullet, $cs, $semantic, $function) = @_;
     #    my ($opensize, $closesize) = (T_CS('\left'), T_CS('\right'));
-    my $cfunc  = Invocation(T_CS('\@CSYMBOL'), $semantic);
-    my $pfunc  = $function;
-    my $inline = $gullet->readMatch(T_OTHER('*'));
-    my $power  = $gullet->readOptional;                      # Optional arg is degree
-    my $tmp1   = $gullet->readArg();                         # 1st arg required
-    my ($tmp2, $open, $close) = phys_readArg($gullet, 0, undef, undef, '(' => T_OTHER(')'));
-    my ($expr, $var) = ($tmp2 ? ($open ? ($tmp2, $tmp1) : ($tmp1, $tmp2)) : (undef, $tmp1));
-    my ($cargs, $pargs) = dualize_arglist('#1#2#3', $expr, $var, $power);
-    return Invocation(T_CS('\DUAL'), undef,
-      # derivative with single arg is operator ?
-      Invocation(T_CS('\@APPLY'), Tokens($cfunc, grep { $_ } @$cargs)),
-      Invocation(T_CS('\@WRAP'),
+    my $semantic_string = ToString($semantic);
+    my $inline          = $gullet->readMatch(T_OTHER('*'));
+    my $power           = $gullet->readOptional;              # Optional arg is degree
+    my $tmp1            = $gullet->readArg();                 # 1st arg required: var1 or expr
```

Comment:
What are `$tmp1` through `$tmp3` used for? Are they input tokens? If so maybe `$tok1` through `$tok3` are better names?

Reply:
they're input arguments, maybe `{}` delimited, maybe `()` delimited, but not single tokens.  They're tmp because the order gets determined by how many of what kind you get.